File: service/src/main/kotlin/es/ubu/lsi/ubumonitorweb/feature/dto/Course.kt
package es.ubu.lsi.ubumonitorweb.feature.dto

interface Course {
  val id: Int
  val fullname: String
  val summary: String?
  val summaryformat: Int?
  val courseimage: String?
  val isfavourite: Boolean?
  val startdate: Long?
  val enddate: Long?

  fun normalize(category: MoodleCategory) =
    NormalizedCourse(
      id = id,
      fullname = fullname,
      summary = summary,
      summaryformat = summaryformat,
      courseimage = courseimage,
      isfavourite = isfavourite,
      startdate = startdate,
      enddate = enddate,
      category = category,
    )
}

File: client-ng/src/app/core/models/course.ts
export enum Format {
  MOODLE = 0,
  HTML = 1,
  PLAIN = 2,
  MARKDOWN = 4,
}

export interface Category {
  id: number;
  name: string;
  description: string;
  descriptionformat: Format;
  path: string;
}

export interface ApiCourse {
  id: number;
  fullname: string;
  summary: string;
  summaryformat: Format;
  courseimage: string;
  isfavourite: boolean;
  startdate: number;
  enddate: number;
  category: Category;
}

export type Course = Omit<ApiCourse, 'startdate'  | 'enddate'> & {
  startdate: Date;
  enddate: Date;
};
